harness/tests/localnet_test.go

```diff
@@ -74,145 +70,78 @@ func TestMain(m *testing.M) {
 func TestSendBidTransaction(t *testing.T) {
 
 	walletsPass := os.Getenv("DUSK_WALLET_PASS")
-	if len(walletsPass) == 0 {
-		t.Logf("Empty DUSK_WALLET_PASS")
-	}
 
 	// Send request to all nodes to loadWallet
 	for i := 0; i < localNetSize; i++ {
-		_, err := localNet.SendCommand(uint(i), "loadwallet", []string{walletsPass})
+		_, err := localNet.LoadWalletCmd(uint(i), walletsPass)
 		if err != nil {
 			t.Fatal(err.Error())
 		}
 	}
 
 	// Send request to node 0 to generate and process a Bid transaction
-	data, err := localNet.SendCommand(0, "bid", []string{"10", "10"})
+	txidBytes, err := localNet.SendBidCmd(0, 10, 10)
 	if err != nil {
 		t.Fatal(err.Error())
 	}
 
-	txid := strings.Trim(data, "Success! TxID: ")
-	t.Logf("Bid transaction id: %s", txid)
+	txID := hex.EncodeToString(txidBytes)
+	t.Logf("Bid transaction id: %s", txID)
 
 	// Ensure all nodes have accepted this transaction at the same height
 	blockhash := ""
-	for i, node := range localNet.Nodes {
-		condition := func() bool {
-			// Construct query to fetch txid
-			query := fmt.Sprintf(
-				"{\"query\" : \"{ transactions (txid: \\\"%s\\\") { txid blockhash } }\"}",
-				txid)
-
-			var resp map[string]map[string][]map[string]string
-			err := localNet.SendQuery(uint(i), query, &resp)
-			if err != nil {
-				return false
-			}
-
-			result, ok := resp["data"]
-			if !ok {
-				return false
-			}
+	for i := 0; i < len(localNet.Nodes); i++ {
 
-			if len(result["transactions"]) == 0 {
-				// graphql request processed but still txid not found
-				return false
-			}
-
-			if i == 0 {
-				blockhash = result["transactions"][0]["blockhash"]
-			}
+		bh := localNet.WaitUntilTx(t, uint(i), txID)
 
-			if len(blockhash) != 0 && blockhash != result["transactions"][0]["blockhash"] {
-				// the case where the network has inconsistency and same tx has been
-				// accepted within different blocks
-				return false
-			}
-
-			if result["transactions"][0]["txid"] != txid {
-				return false
-			}
+		if len(bh) == 0 {
+			t.Fatal("empty blockhash")
+		}
 
-			return true
+		if len(blockhash) != 0 && blockhash != bh {
+			// the case where the network has inconsistency and same tx has been
+			// accepted within different blocks
+			t.Fatal("same tx hash has been accepted within different blocks")
 		}
 
-		// asserts that given condition will be met in 1 minute, by checking condition function each second.
-		if !assert.Eventuallyf(t, condition, 1*time.Minute, time.Second, "failed node %s", node.Id) {
-			break
+		if i == 0 {
+			blockhash = bh
 		}
 	}
 }
 
 // TestCatchup tests that a node which falls behind during consensus
 // will properly catch up and re-join the consensus execution trace.
 func TestCatchup(t *testing.T) {
+
```

Comment:
@julesdesmit Let me know if the refactored TestCatchup works for you as I tried to avoid the need of "PublishTopic" rpc call. If not, I could restore the initial idea test and add `PublishTopic` to gRPC node service.

Reply:
Yep, both tests are passing without issue on my end.